activerecord/lib/active_record/model_schema.rb

```diff
@@ -574,7 +575,6 @@ def reload_schema_from_cache
           @content_columns = nil
           @default_attributes = nil
           @column_defaults = nil
-          @inheritance_column = nil unless defined?(@explicit_inheritance_column) && @explicit_inheritance_column
```

Comment:
This `@explicit_inheritance_column` was useless since it would never cache the parent values but simply default to `nil`.